File: game.js
const WIDTH = 256;
const HEIGHT = 256;

function Game(vm) {

    return {
        tick: (deltaTime) => {
            vm.tmap('tilemap');

            for (let i = 0; i < 16; ++i) {
                vm.stile(i, 0);
                vm.spos(i, i * 16, HEIGHT - 16);
            }

            vm.stile(16, 2);
            vm.spos(16, 3 * 16, HEIGHT - 48);
            vm.stile(17, 3);
            vm.spos(17, 4 * 16, HEIGHT - 48);
            vm.stile(18, 4);
            vm.spos(18, 3 * 16, HEIGHT - 32);
            vm.stile(19, 5);
            vm.spos(19, 4 * 16, HEIGHT - 32);
        }
    }
}

function boot(vm) {
    return Game(vm);
}

exports.boot = boot;

File: game.lua
mode(MDEFAULT)
-- select tilemap.png as the tilemap
tmap('tilemap')
-- assign the tile 0 to sprite 0
stile(0, 6)
-- origin of sprite 0 is the center
sorigin(0, 8, 8)
-- center sprite 0 on screen
spos(0, SWIDTH / 2, SHEIGHT / 2);

function tick(deltaTime)
    -- rotate sprite 0
    srot(0, srot(0) + deltaTime)
    if (btn(BLEFT)) then
        x, y = sscale(0)
        sscale(0, x + deltaTime / 10, y + deltaTime / 10)
    end
end
